tilelang/utils/language.py

```
@@ -131,8 +131,16 @@ def get_buffer_region_from_load(buffer_load: tir.BufferLoad) -> Optional[tir.Buf
     for indice in indices:
-        if not isinstance(indice, tir.Ramp):
-            return None
-        regions.append(ir.Range.from_min_extent(indice.base, indice.lanes))
-    return tir.BufferRegion(buffer, regions)
+        if isinstance(indice, tir.Ramp):
+            regions.append(ir.Range.from_min_extent(indice.base, indice.lanes))
+            found_ramp = True
+        elif isinstance(indice, tir.PrimExpr):
+            regions.append(ir.Range.from_min_extent(indice, 1))
+        else:
+            raise ValueError("Unsupported type: ", type(indice))
```

Ruff (0.13.3)

Prefer `TypeError` exception for invalid type

(TRY004)